client/services/FeedbackGeneratorService.js

```diff
@@ -30,6 +30,33 @@ tie.factory('FeedbackGeneratorService', [
     RUNTIME_ERROR_FEEDBACK_MESSAGES, WRONG_LANGUAGE_ERRORS, LANGUAGE_PYTHON,
     CLASS_NAME_AUXILIARY_CODE, CLASS_NAME_SYSTEM_CODE, PARAGRAPH_TYPE_TEXT,
     PARAGRAPH_TYPE_CODE, PARAGRAPH_TYPE_SYNTAX_ERROR) {
+    /**
+     *
+     */
+    var UNFAMILIARITY_THRESHOLD = 5;
+
+    var consecutiveSyntaxErrorCounter = 0;
+    var consecutiveSameErrorCounter = 0;
+    var consecutiveWrongLanguageCounter = 0;
+    var previousErrorString = '';
+
+    /**
+     *
+     */
+    var _resetCounters = function(currentCtr) {
+      var notResetCtr = currentCtr || '';
+      if (notResetCtr !== 'consecutiveSyntaxErrorCounter') {
```

Comment:
I get the feeling there is a better way to do this rather than comparing direct strings that can be more precise and foolproof. Perhaps we can use constants to represent these strings?